solver/llbsolver/file/backend.go

```diff
@@ -190,14 +184,10 @@ func rmPath(root, src string, allowNotFound bool) error {
 }
 
 func docopy(ctx context.Context, src, dest string, action *pb.FileActionCopy, u *copy.User, idmap *idtools.IdentityMapping) (err error) {
-	srcPath, err := cleanPath(action.Src)
-	if err != nil {
-		return errors.Wrap(err, "cleaning source path")
-	}
-	destPath, err := cleanPath(action.Dest)
-	if err != nil {
-		return errors.Wrap(err, "cleaning destination path")
-	}
+	// NB: the paths do not need cleaning since they are always in unix style
```

Comment:
Looking at it again. The old function seems to also collapse the `..` relative paths with `Join("/", )` . I guess we still need that functionality.

Reply:
Ok, I just retain this part since we don't need ` system.CheckSystemDriveAndRemoveDriveLetter` this point?

```go
func cleanPath(s string) (string, error) {
	if err != nil {
		return "", errors.Wrap(err, "removing drive letter")
	}
	s = filepath.FromSlash(s)
	s2 := filepath.Join("/", s)
	if strings.HasSuffix(s, string(filepath.Separator)+".") {
		if s2 != string(filepath.Separator) {
			s2 += string(filepath.Separator)
		}
		s2 += "."
	} else if strings.HasSuffix(s, string(filepath.Separator)) && s2 != string(filepath.Separator) {
		s2 += string(filepath.Separator)
	}
	return s2, nil
}
```

Reply:
Or if you just use `path.` then you don't need `FromSlash` I guess.